### PR TITLE
CHANGELOG: Add a raw version with all PRs towards Mixxx 2.4  

### PR DESCRIPTION
**Pull request by daschuer**
This is an important step towards the long overdue 2.4 release. We have already integrated ~800 PRs .. Thank you all for this hard work. 

The rendered Changelog looks like this: 
https://github.com/daschuer/mixxx/blob/9cfe9e4d0f8eacafc4c82be87a352e48c3ad167b/CHANGELOG.md

I like to keep this full list as a commit for later reference, documenting our next meld down step. 
So please have a look. 

The next step would be cut out a 2.4 branch. I am still very much in favor or a 6 month release cycle to stop wining about a view very last PRs that should become part of a certain release and than delay all the other 800 by another six month or such.  

Let's now also collect PRs that should still go to 2.4 at the 2.4 Milestone:
https://github.com/mixxxdj/mixxx/pulls?q=is%3Aopen+is%3Apr+milestone%3A2.4.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -96,8 +101,17 @@
...
+* Recording: Fix table refresh issues [#4648](https://github.com/mixxxdj/mixxx/pull/4648)
+* Add popup before deletion playlist and crates [#4697](https://github.com/mixxxdj/mixxx/pull/4697) [#4700](https://github.com/mixxxdj/mixxx/pull/4700) [lp:1912475](https://bugs.launchpad.net/mixxx/+bug/1912475)
+* Experimental: Fix writing of undefined MusicBrainz Recording ID   [#4694](https://github.com/mixxxdj/mixxx/pull/4694)
+* Traktor library: fix importing track key [#4701](https://github.com/mixxxdj/mixxx/pull/4701)
+* Add History cleanup options [#4726](https://github.com/mixxxdj/mixxx/pull/4726) [lp:1766163](https://bugs.launchpad.net/mixxx/+bug/1766163) [lp:1969217](https://bugs.launchpad.net/mixxx/+bug/1969217)
+* Track Menu: Add ability to select loaded track in library [#4740](https://github.com/mixxxdj/mixxx/pull/4740)
+* Add track context menu entry for Analyze and Reanlyze [#4806](https://github.com/mixxxdj/mixxx/pull/4806)
+* Track Properties: Show 'date added' as local time [#4838](https://github.com/mixxxdj/mixxx/pull/4838) [lp:1980658](https://bugs.launchpad.net/mixxx/+bug/1980658)
```
Comment (ronso0) [inline CHANGELOG.md:114]: this went into 2.3

**Comment by ronso0** [comment]
+ Implement Un-eject to reload the previous track [4868](https://github.com/mixxxdj/mixxx/pull/4668)

File: CHANGELOG.md [hunk 2]
```diff
@@ -255,18 +285,27 @@
...
+* Behringer DDM4000 mixer: Update controller mapping [#4262](https://github.com/mixxxdj/mixxx/pull/4262) [#4799](https://github.com/mixxxdj/mixxx/pull/4799)
 * Numark DJ2GO2 Touch: Fix sampler, hotcue, beatloop buttons [#4287](https://github.com/mixxxdj/mixxx/pull/4287)
+* Numark DJ2GO2 Touch: Fix Touch sliders and knobs [#4835](https://github.com/mixxxdj/mixxx/pull/4835) [lp:1948596](https://bugs.launchpad.net/mixxx/+bug/1948596)
 * Denon MC6000MK2: Improve mapping code [#4385](https://github.com/mixxxdj/mixxx/pull/4385)
 * Yaeltex MiniMixxx: Add controller mapping [#4350](https://github.com/mixxxdj/mixxx/pull/4350)
+* Added Trace for the mapping connections, to allow JS profiling [#4766](https://github.com/mixxxdj/mixxx/pull/4766)
+* Fix crashe when using midi.sendShortMsg and platform vnc [#4635](https://github.com/mixxxdj/mixxx/pull/4635)
+* Traktor S3: Fix timedelta calculation bugs [#4646](https://github.com/mixxxdj/mixxx/pull/4646) [lp:1958925](https://bugs.launchpad.net/mixxx/+bug/1958925)
+* Pot support for arbitrary maximums in 7-/14-bit handlers from controller scripts[#4495](https://github.com/mixxxdj/mixxx/pull/4495)
```
Comment (ronso0) [inline CHANGELOG.md:308]:
```suggestion
* add support to Potmeters to  handler arbitrary maximums in 7-/14-bit from controller scripts [#4495](https://github.com/mixxxdj/mixxx/pull/4495)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -338,18 +397,30 @@
...
 * Allow skin scaling from preferences [#3960](https://github.com/mixxxdj/mixxx/pull/3960)
 * Invert scroll wheel waveform zoom direction to mach other applications [#4195](https://github.com/mixxxdj/mixxx/pull/4195)
+* Waveform: Avoid visual glitch with ranges < 1 px [#4804](https://github.com/mixxxdj/mixxx/pull/4804)
 * Fix crash if no skin is available
   [#3918](https://github.com/mixxxdj/mixxx/pull/3918)
   [#3939](https://github.com/mixxxdj/mixxx/pull/3939)
 * Fix leaked controls [#4213](https://github.com/mixxxdj/mixxx/pull/4213) [lp:1912129](https://bugs.launchpad.net/mixxx/+bug/1912129)
-* Shade: remove initial setting of now accessible effect controls [#4398](https://github.com/mixxxdj/mixxx/pull/4398) [lp:1946811](https://bugs.launchpad.net/mixxx/+bug/1946811)
+* Shade: Remove initial setting of now accessible effect controls [#4398](https://github.com/mixxxdj/mixxx/pull/4398) [lp:1946811](https://bugs.launchpad.net/mixxx/+bug/1946811)
+* Shade: Fix library sidebar splitter glitch [#4828](https://github.com/mixxxdj/mixxx/pull/4828) [lp:1979823](https://bugs.launchpad.net/mixxx/+bug/1979823)
```
Comment (ronso0) [inline CHANGELOG.md:406]: this was backported to 2.3.4 #4808

File: CHANGELOG.md [hunk 4]
```diff
@@ -338,18 +397,30 @@
 * Make beat indicator control behaviour more natural [#3608](https://github.com/mixxxdj/mixxx/pull/3608)
 * Allow skin scaling from preferences [#3960](https://github.com/mixxxdj/mixxx/pull/3960)
 * Invert scroll wheel waveform zoom direction to mach other applications [#4195](https://github.com/mixxxdj/mixxx/pull/4195)
+* Waveform: Avoid visual glitch with ranges < 1 px [#4804](https://github.com/mixxxdj/mixxx/pull/4804)
```
Comment (ronso0) [inline CHANGELOG.md:400]: went into 2.3.4

File: CHANGELOG.md [hunk 5]
```diff
@@ -338,18 +397,30 @@
...
+* Shade: Fix library sidebar splitter glitch [#4828](https://github.com/mixxxdj/mixxx/pull/4828) [lp:1979823](https://bugs.launchpad.net/mixxx/+bug/1979823)
 * Fix switching from Shade to other skins [#4421](https://github.com/mixxxdj/mixxx/pull/4421) [lp:1946812](https://bugs.launchpad.net/mixxx/+bug/1946812)
 * Use double click to reset knobs and sliders [#4509](https://github.com/mixxxdj/mixxx/pull/4509) [lp:1875999](https://bugs.launchpad.net/mixxx/+bug/1875999)
 * Use info not warning for skin COs [#4525](https://github.com/mixxxdj/mixxx/pull/4525)
+* LateNight: move logo to the right [#4677](https://github.com/mixxxdj/mixxx/pull/4677)
+* Spinny: allow to toggle cover art at runtime [#4565](https://github.com/mixxxdj/mixxx/pull/4565) [lp:1883362](https://bugs.launchpad.net/mixxx/+bug/1883362)
+* Add new searchbox styles [#4505](https://github.com/mixxxdj/mixxx/pull/4505)
+* Deere: add QuickEffect selector [#4590](https://github.com/mixxxdj/mixxx/pull/4590)
+* LateNight: add effect chain menu button [#4593](https://github.com/mixxxdj/mixxx/pull/4593)
```
Comment (ronso0) [inline CHANGELOG.md:414]:
```suggestion
* add new Quick Effect selector and effect chain buttons [#4501](https://github.com/mixxxdj/mixxx/pull/4501) [#4502](https://github.com/mixxxdj/mixxx/pull/4502) [#4503](https://github.com/mixxxdj/mixxx/pull/4503) [#4590](https://github.com/mixxxdj/mixxx/pull/4590)  [#4593](https://github.com/mixxxdj/mixxx/pull/4593)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -375,6 +446,11 @@
   [#4461](https://github.com/mixxxdj/mixxx/pull/4461)
   [#4548](https://github.com/mixxxdj/mixxx/pull/4548)
   [#4503](https://github.com/mixxxdj/mixxx/pull/4503)
+  [#4686](https://github.com/mixxxdj/mixxx/pull/4686)
+  [#4691](https://github.com/mixxxdj/mixxx/pull/4691)
+  [#4704](https://github.com/mixxxdj/mixxx/pull/4704)
+  [#4748](https://github.com/mixxxdj/mixxx/pull/4748)
+  [#4833](https://github.com/mixxxdj/mixxx/pull/4833) [lp:1980078](https://bugs.launchpad.net/mixxx/+bug/1980078)
```
Comment (ronso0) [inline CHANGELOG.md:453]: indentation here & above

File: CHANGELOG.md [hunk 7]
```diff
@@ -246,6 +274,8 @@
   [#4217](https://github.com/mixxxdj/mixxx/pull/4217)
   [#1966](https://github.com/mixxxdj/mixxx/pull/1966)
   [#4535](https://github.com/mixxxdj/mixxx/pull/4535)
+  [#4770](https://github.com/mixxxdj/mixxx/pull/4770)
+  [#4771](https://github.com/mixxxdj/mixxx/pull/4771)
```
[the remark this answers is not in the document]
Comment (daschuer) [reply CHANGELOG.md:277]: I am unclear what is the issue here. Is it solved already?

**Comment by daschuer** [comment]
I have moved all 2.3.4 changes to https://github.com/mixxxdj/mixxx/pull/4842

**Comment by daschuer** [comment]
This is now rebased on top of current main including the unrelease 2.3.4 changelog.

**Comment by daschuer** [comment]
@ronso0 this should now also be up to date.

File: CHANGELOG.md [hunk 8]
```diff
@@ -106,11 +109,12 @@
...
-* Experimental: Fix writing of undefined MusicBrainz Recording ID   [#4694](https://github.com/mixxxdj/mixxx/pull/4694)
+* Experimental: Fix writing of undefined MusicBrainz Recording ID [#4694](https://github.com/mixxxdj/mixxx/pull/4694)
 * Traktor library: fix importing track key [#4701](https://github.com/mixxxdj/mixxx/pull/4701)
 * Add History cleanup options [#4726](https://github.com/mixxxdj/mixxx/pull/4726) [lp:1766163](https://bugs.launchpad.net/mixxx/+bug/1766163) [lp:1969217](https://bugs.launchpad.net/mixxx/+bug/1969217)
 * Track Menu: Add ability to select loaded track in library [#4740](https://github.com/mixxxdj/mixxx/pull/4740)
-* Add track context menu entry for Analyze and Reanlyze [#4806](https://github.com/mixxxdj/mixxx/pull/4806)
+* Add a track context menu entry for Analyze and Reanlyze [#4806](https://github.com/mixxxdj/mixxx/pull/4806)
+* Add a track context menu for looking up track metadata at Discogs, Soundcoud an LastFM [#4772](https://github.com/mixxxdj/mixxx/pull/4772) [#4836](https://github.com/mixxxdj/mixxx/pull/4836)
```
Comment (ronso0) [inline CHANGELOG.md:117]:
```suggestion
* Add a track context menu for looking up track metadata at Discogs, Soundcoud and LastFM [#4772](https://github.com/mixxxdj/mixxx/pull/4772) [#4836](https://github.com/mixxxdj/mixxx/pull/4836)
```

File: CHANGELOG.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (ronso0) [reply CHANGELOG.md:281]: I guess it doesn't matter, just noticed it

**Comment by daschuer** [comment]
Anyone else want that want to take a look?

File: CHANGELOG.md [hunk 9]
```diff
@@ -96,8 +104,17 @@
...
+* Recording: Fix table refresh issues [#4648](https://github.com/mixxxdj/mixxx/pull/4648)
+* Add popup before deletion playlist and crates [#4697](https://github.com/mixxxdj/mixxx/pull/4697) [#4700](https://github.com/mixxxdj/mixxx/pull/4700) [lp:1912475](https://bugs.launchpad.net/mixxx/+bug/1912475)
+* Experimental: Fix writing of undefined MusicBrainz Recording ID [#4694](https://github.com/mixxxdj/mixxx/pull/4694)
+* Traktor library: fix importing track key [#4701](https://github.com/mixxxdj/mixxx/pull/4701)
+* Add History cleanup options [#4726](https://github.com/mixxxdj/mixxx/pull/4726) [lp:1766163](https://bugs.launchpad.net/mixxx/+bug/1766163) [lp:1969217](https://bugs.launchpad.net/mixxx/+bug/1969217)
+* Track Menu: Add ability to select loaded track in library [#4740](https://github.com/mixxxdj/mixxx/pull/4740)
+* Add a track context menu entry for Analyze and Reanlyze [#4806](https://github.com/mixxxdj/mixxx/pull/4806)
+* Add a track context menu for looking up track metadata at Discogs, Soundcoud and LastFM [#4772](https://github.com/mixxxdj/mixxx/pull/4772) [#4836](https://github.com/mixxxdj/mixxx/pull/4836)
```
Comment (ronso0) [inline CHANGELOG.md:117]:
```suggestion
* Track Menu: Add action to select loaded track in library [#4740](https://github.com/mixxxdj/mixxx/pull/4740)
* Track Menu: Add menu for Analyze and Reanlyze [#4806](https://github.com/mixxxdj/mixxx/pull/4806)
* Track Menu: Add menu for looking up track metadata at Discogs, Soundcoud and LastFM [#4772](https://github.com/mixxxdj/mixxx/pull/4772) [#4836](https://github.com/mixxxdj/mixxx/pull/4836)
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -152,6 +172,15 @@
 * Fix type detection of AIFF files [#4364](https://github.com/mixxxdj/mixxx/pull/4364)
 * AAC encoder: Fixed a memory leak [#4386](https://github.com/mixxxdj/mixxx/pull/4386) [#4408](https://github.com/mixxxdj/mixxx/pull/4408)
 * Improve robustness of file type detection by considering the actual MIME type of the content. [lp:1445885](https://bugs.launchpad.net/mixxx/+bug/1445885) [#4356](https://github.com/mixxxdj/mixxx/pull/4356) [#4357](https://github.com/mixxxdj/mixxx/pull/4357)
+* Fix file type detection when file has wrong file extension by determining the MIME type from content [#4602](https://github.com/mixxxdj/mixxx/pull/4602) [#4600](https://github.com/mixxxdj/mixxx/pull/4600)
+* Fix handling of files with wrong suffix
+  [#4615](https://github.com/mixxxdj/mixxx/pull/4615)
+  [lp:1445885](https://bugs.launchpad.net/mixxx/+bug/1445885)
+  [#4589](https://github.com/mixxxdj/mixxx/pull/4589)
```
Comment (ronso0) [inline CHANGELOG.md:179]: #4589 shouldn't this be in the **Audio Engine** category?

File: CHANGELOG.md [hunk 11]
```diff
@@ -172,6 +201,7 @@
 * Fix/Improve snapping to beats in various situations [#4366](https://github.com/mixxxdj/mixxx/pull/4366) [lp:1945238](https://bugs.launchpad.net/mixxx/+bug/1945238)
 * Allow to cancel active loops via beatloop_activate [#4328](https://github.com/mixxxdj/mixxx/pull/4328) [lp:1876003](https://bugs.launchpad.net/mixxx/+bug/1876003)
 * Don't wipe inapplicable sound config immediately [#4544](https://github.com/mixxxdj/mixxx/pull/4544)
+* Allow to reset loops via "[ChannelN], loop_remove" control object [#4802](https://github.com/mixxxdj/mixxx/pull/4802) [lp:1978507](https://bugs.launchpad.net/mixxx/+bug/1978507)
```
Comment (ronso0) [inline CHANGELOG.md:204]: This should go to below the other **Track Menu** entries, though just to improve discoverability?
Or we add it twice: the track menu action to **Library** and the control here?

File: CHANGELOG.md [hunk 12]
```diff
@@ -255,18 +288,25 @@
...
 * Components JS: Fix syncbutton [#4329](https://github.com/mixxxdj/mixxx/pull/4329)
 * Roland DJ-505: Make blinking lights blink in sync and other improvements [#4159](https://github.com/mixxxdj/mixxx/pull/4159) [#4517](https://github.com/mixxxdj/mixxx/pull/4517)
-* Behringer DDM4000 mixer: Update controller mapping [#4262](https://github.com/mixxxdj/mixxx/pull/4262)
+* Behringer DDM4000 mixer: Update controller mapping [#4262](https://github.com/mixxxdj/mixxx/pull/4262) [#4799](https://github.com/mixxxdj/mixxx/pull/4799)
 * Numark DJ2GO2 Touch: Fix sampler, hotcue, beatloop buttons [#4287](https://github.com/mixxxdj/mixxx/pull/4287)
 * Denon MC6000MK2: Improve mapping code [#4385](https://github.com/mixxxdj/mixxx/pull/4385)
 * Yaeltex MiniMixxx: Add controller mapping [#4350](https://github.com/mixxxdj/mixxx/pull/4350)
+* Added Trace for the mapping connections, to allow JS profiling [#4766](https://github.com/mixxxdj/mixxx/pull/4766)
+* Fix crashe when using midi.sendShortMsg and platform vnc [#4635](https://github.com/mixxxdj/mixxx/pull/4635)
```
Comment (ronso0) [inline CHANGELOG.md:308]:
```suggestion
* Fix crash when using `midi.sendShortMsg` and platform vnc [#4635](https://github.com/mixxxdj/mixxx/pull/4635)
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -342,15 +402,25 @@
...
 * Use double click to reset knobs and sliders [#4509](https://github.com/mixxxdj/mixxx/pull/4509) [lp:1875999](https://bugs.launchpad.net/mixxx/+bug/1875999)
 * Use info not warning for skin COs [#4525](https://github.com/mixxxdj/mixxx/pull/4525)
+* LateNight: Move logo to the right [#4677](https://github.com/mixxxdj/mixxx/pull/4677)
+* Spinny: Allow to toggle cover art at runtime [#4565](https://github.com/mixxxdj/mixxx/pull/4565) [lp:1883362](https://bugs.launchpad.net/mixxx/+bug/1883362)
+* Add new searchbox styles [#4505](https://github.com/mixxxdj/mixxx/pull/4505)
+* Add new Quick Effect selector and effect chain buttons [#4501](https://github.com/mixxxdj/mixxx/pull/4501) [#4502](https://github.com/mixxxdj/mixxx/pull/4502) [#4503](https://github.com/mixxxdj/mixxx/pull/4503) [#4590](https://github.com/mixxxdj/mixxx/pull/4590)  [#4593](https://github.com/mixxxdj/mixxx/pull/4593)
+* Fix crash when starting without a valid skin directory [#4575](https://github.com/mixxxdj/mixxx/pull/4575) [1934560](https://bugs.launchpad.net/mixxx/+bug/1934560)
```
Comment (ronso0) [inline CHANGELOG.md:413]:
```suggestion
* Fix crash when starting without a valid skin directory [#4575](https://github.com/mixxxdj/mixxx/pull/4575) [lp:1934560](https://bugs.launchpad.net/mixxx/+bug/1934560)
```

**Review by ronso0 — CHANGES_REQUESTED** [review]
Thank you for doing this tedious job!

Some typos and other minor remarks.
Note I only checked Changelog.md, someone else may do the final check that meatinfo.xml matches

File: CHANGELOG.md [hunk 14]
```diff
@@ -1023,6 +1185,7 @@
 * Fix quoted search sentence. [lp:1784141](https://bugs.launchpad.net/mixxx/+bug/1784141)
 * Fix loading a track formerly not existing. [lp:1800395](https://bugs.launchpad.net/mixxx/+bug/1800395)
 * Fix importing m3u files with blank lines. [lp:1806271](https://bugs.launchpad.net/mixxx/+bug/1806271)
+* Fix exporting m3u files with tracks and special characters by using the URL format [#4752](https://github.com/mixxxdj/mixxx/pull/4752)
```
Comment (ronso0) [inline CHANGELOG.md:1188]: how did this end up in the 2.1.6 changelog? : )

File: CHANGELOG.md [hunk 15]
```diff
@@ -834,6 +995,7 @@
 * Add Opus streaming and recording support. [lp:1338413](https://bugs.launchpad.net/mixxx/+bug/1338413)
 * Remove support for SoundSource plugins because the code was not well-maintained and could lead to crashes [lp:1792747](https://bugs.launchpad.net/mixxx/+bug/1792747)
 * Add HE-AAC encoding capabilities for recording and broadcasting [#3615](https://github.com/mixxxdj/mixxx/pull/3615)
+* ID3v2 parsing: Improve log warnings [#4610](https://github.com/mixxxdj/mixxx/pull/4610)
```
Comment (ronso0) [inline CHANGELOG.md:998]: also move to **2.4**

**Review by ronso0 — CHANGES_REQUESTED** [review]
entries that need to be moved to **2.4**

**Comment by Swiftb0y** [comment]
> Note I only checked Changelog.md, someone else may do the final check that meatinfo.xml matches

Not necessary IMO. The metainfo.xml is generated automatically from the changelog.

**Comment by daschuer** [comment]
Done

**Comment by ronso0** [comment]
There are some issues left
https://github.com/mixxxdj/mixxx/pull/4840#pullrequestreview-1061560881

**Comment by ronso0** [comment]
Thank you :tada:

**Review by ronso0 — APPROVED** [review]

**Comment by daschuer** [comment]
Thank you very much for your thoughtful review!

**Pull request merged**